src/design_parameters.rs

```diff
@@ -7,3 +7,12 @@ pub const ADC_DAC_SCK_MHZ_MAX: u32 = 50;
 
 /// The optimal counting frequency of the hardware timers used for timestamping and sampling.
 pub const TIMER_FREQUENCY_MHZ: u32 = 100;
+
+/// The DDS reference clock frequency in MHz.
+pub const DDS_REF_CLK_MHZ: u32 = 100;
+
+/// The multiplier used for the DDS reference clock PLL.
+pub const DDS_MULTIPLIER: u8 = 5;
+
+/// The rate of the DDS SYNC_CLK in MHz is always 1/4 that of the internal PLL clock.
+pub const DDS_SYNC_CLK_MHZ: u32 = DDS_REF_CLK_MHZ * DDS_MULTIPLIER as u32 / 4;
```

Comment:
I'd refactor this as `DDS_SYNC_CLK_DIV: u32 = 4`. And we only need that once below. So it could be inlined.

Reply:
I believe since these are all `const`, they should actually be evaluated at compile-time, and thus always inlined. I'll go ahead and add a definition though, since it will make it clear where the magic number is coming from.

Reply:
I'm not sure that the result will always be the same (independent of whether this is compile-time or run-time). Operator precedence can lead to different rounding with f32. As explained elsewhere it's best work in integer multipliers and dividers as much as possible and only convert to f32 where really needed. E.g. for the timestamping the only intrinsic `f32` is `pub const DDS_REF_PER_TIMER: f32 = DDS_REF_MHZ/TIMER_CLK_MHz`. The rest are fractions/multiples of that.

Reply:
I don't know if I'm following you here. All the numbers in this are unsigned integers - there's no floating point operation being performed. By substituting in the constants, you can see that the calculation is:
```rust
pub const DDS_SYNC_CLK_MHZ: u32 = 100_u32 * 5_u8 as u32 / 4
```

Writing that out, I'm noticing there's ambiguity in the integer math, so I'm adding a parenthesis (although Rust guarantees left-> right analysis of * and / operators)

Reply:
I was comparing using the multipliers/dividers as integers (and avoiding the calculation and comparison of new SI frequencies/periods) with rounding. Yes it's not f32. In that regard i32 MHz is even worse than f32 Hz. It would be very wrong for e.g.  `DDS_REF_CLK_MHZ = 25`, `DDS_MULTIPLIER=28`.

Reply:
I wasn't really following still until I went and looked at the code fora  refactor and I now see what you are getting at. I've updated it to not calculate the SYNC_CLK using integer math - it's now offloaded into a float before any division occurs.

Reply:
Closing as this should now be resolved. Thanks for pointing it out